client/src/components/screens/PatientSubmission/ClinicalInformation.jsx

```diff
@@ -416,14 +416,20 @@ class ClinicalInformation extends React.Component {
 
   handleHpoNodesChecked(_, info) {
     const { actions, observations } = this.props;
+    const { treeData } = this.state;
 
     const checkedNodes = info.checkedNodes.map(n => ({ code: n.key, display: n.props.title }));
     const hpoResources = observations.hpos;
 
     const toDelete = [];
     const toAdd = [];
     hpoResources.forEach((resource) => {
-      if (checkedNodes.find(r => r.code === resource.valueCodeableConcept.coding[0].code) == null) {
+      const code = resource.valueCodeableConcept.coding[0];
+      const isUnchecked = checkedNodes.find(r => r.code === code) === null;
+      // Resources selected by the autocomplete aren't in the treeData
+      const isHidden = treeData.find(td => td.key === code) === null;
```

Comment:
Also, if no element satisfies the condiition of `Array.find`, it returns `undefined` and not `null`, so testing for `=== null` will always be false. You could use the double equality here `== null` or just test for undefined `=== undefined`.
Check https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find for more info